### DISCUSSION TITLE
Is there other choices about ALPN in sample.c？

### DISCUSSION THREAD

**Discussion by AvatarGanymede** [microsoft/msquic#2036] [Q&A] [answered]
In the sample.c, the `Alpn` is defined as below:
```
const QUIC_BUFFER Alpn = { sizeof("sample") - 1, (uint8_t*)"sample" };
```
If I change "sample" to other strings, the connection would shut by transport, raising `ERROR_QUIC_ALPN_NEG_FAILURE`

My question is: except for "sample", can I choose other "configs"?

**Comment by AvatarGanymede** [comment, marked as answer]
Oh, maybe I know. The string is like a 'ticket' for handshake, the server and the client must share the same ALPN string, right?

**Reply by nibanks** [reply]
Yes, ALPN needs to match. When a server registers a listener for ALPN X, it's like saying "I support protocol X!" and when the client uses it to try to connect, it's saying "Hey, can we use protocol X to talk?".
